Article : création de cartes avec Blender (traduction) (#1107)
geotribu/website: content/articles/2024/2024-04-14_creer_carte_blender.md

---
title: "Réaliser des cartes avec Blender"
subtitle: Blindé jusqu’au relief
authors:
    - Thomas Szczurek-Gayant
categories:
    - article
comments: true
date: 2024-04-07
description: "Technique permettant de réaliser des cartes de relief avec le logiciel libre 3d Blender, ainsi qu'un petit tutoriel GDAL"
icon: simple/blender
image:
license: default
robots: index, follow
tags:
    - 3D
    - Blender
    - cartographie
    - GDAL
---

# Réaliser des cartes avec Blender

(et aussi un petit tuto GDAL en lignes de commandes)

> Cet article se base sur celui initialement paru en anglais sur [somethingaboutmaps](https://somethingaboutmaps.wordpress.com/2017/11/16/creating-shaded-relief-in-blender/), dont on remerciera l'auteur pour l'article initial et l'aimable autorisation d'écrire celui-ci.

Reviewer: Éventuellement citer le nom de l'auteur original du tuto? (Daniel Huffman) 
Ça personnifie l'histoire, et ça lui donne un peu plus de visibilité :)

Ou comment faire des cartes qui ont la classe. Grossièrement, la technique consiste à déformer un plan avec un raster d'élévation.

> Je vais ici utiliser le  MNT à 1 mètre issus du [RGE ALTI](https://geoservices.ign.fr/rgealti) de l'IGN sur le département des Pyrennées Atlantiques, ce qui permettra d'avoir et du relief (ce qui rend bien avec cette technique) et de la mer (sacro-sainte règle des effets de manche pour avoir la classe : mettez de la flotte).

Reviewer: ```suggestion
Je vais ici utiliser le  MNT à 1 mètre issus du [RGE ALTI](https://geoservices.ign.fr/rgealti) de l'IGN sur le département des Pyrénées-Atlantiques, ce qui permettra d'avoir du relief (ce qui rend bien avec cette technique) et de la mer (sacro-sainte règle des effets de manche pour avoir la classe : mettez de la flotte).
```

> Il y a plusieurs étapes de préparation des données et l'une d'entre elle nécessite obligatoirement l'utilisation de GDAL en lignes de commande. Restez ici ! Rien de bien compliqué et on vous explique tout (et on va profiter de cet article pour essayer de faire tout les pré-traitmements raster en lignes de commande pour s'y familiariser. Au cazou j'indiquerai aussi comment faire avec Qgis).

Reviewer: ```suggestion
Il y a plusieurs étapes de préparation des données et l'une d'entre elles nécessite obligatoirement l'utilisation de GDAL en ligne de commande. Restez ici ! Rien de bien compliqué et on vous explique tout (et on va profiter de cet article pour essayer de faire tout les pré-traitements raster en lignes de commande pour s'y familiariser. Au cazou j'indiquerai aussi comment faire avec QGIS).
```

> Ceci implique d'avoir accès à GDAL. Sur windows ça passe en se rendant dans votre répertoire d'installation de Qgis et ensuite en démarrant OSGeo4W.bat. Je considère que les linuxiens sont assez aguéris pour se débrouiller (au pire faites vous un environnement [mamba](https://mamba.readthedocs.io/en/latest/) qui va bien (miniconda réécrit en c/c++)) et je refuse par principe de parler aux appeliens (sauf à ma cheffe de service car je suis bien obligé).

Reviewer: ```suggestion
Ceci implique d'avoir accès à GDAL. Sur Windows, ça se passe en se rendant dans votre répertoire d'installation de QGIS, puis en démarrant OSGeo4W.bat. Je considère que les linuxiens sont assez aguerris pour se débrouiller (au pire faites-vous un environnement [mamba](https://mamba.readthedocs.io/en/latest/) qui va bien (miniconda réécrit en c/c++)) et je refuse par principe de parler aux apple-iens (sauf à ma cheffe de service car je suis bien obligé).
```

J'ai remplacé appeliens par apple-iens, ça me semblait un peu plus lisible du premier coup!

## Préparation des données

> Après récupération du jeu de données, chargez une des dalles au format asc dans qgis. Le format asc ne gère pas les projections et Qgis essayera par défaut de les positionner en WGS 84 (ne vous inquietez pas, les coordonnées sont les bonnes). Vous pouvez repositionner l'image en cliquant sur la petite icône de l'image ci-dessous et en spécifiant ce bon vieux Lambert 93.

Reviewer: ```suggestion
Après récupération du jeu de données, chargez une des dalles au format ASC dans qgis. Le format ASC ne gère pas les projections et QGIS essayera par défaut de les positionner en WGS84 (ne vous inquiétez pas, les coordonnées sont les bonnes). Vous pouvez repositionner l'image en cliquant sur la petite icône de l'image ci-dessous et en spécifiant ce bon vieux Lambert 93.
```

> ![Ca arrive même aux meilleurs](https://cdn.geotribu.fr/img/articles-blog-rdp/articles/2024/gdal_qgis_blender/img1_lign.png){: .img-center loading=lazy }

Reviewer: ```suggestion
![Ça arrive même aux meilleurs](https://cdn.geotribu.fr/img/articles-blog-rdp/articles/2024/gdal_qgis_blender/img1_lign.png){: .img-center loading=lazy }
```

> Sur ces considérations on chargera plutôt la couche "dalles" [dans le format dont il ne faut pas prononcer le nom](http://switchfromshapefile.org/) située dans le dossier "3_SUPPLEMENTS_LIVRAISON" de l'archive et on active un fond de plan open street map sous "xyz tiles" de l'explorateur de Qgis histoire de se repérer.

Reviewer: ```suggestion
Sur ces considérations on chargera plutôt la couche "dalles" [dans le format dont il ne faut pas prononcer le nom](http://switchfromshapefile.org/) située dans le dossier "3_SUPPLEMENTS_LIVRAISON" de l'archive. On active ensuite un fond de plan OpenStreetMap sous "xyz tiles" de l'explorateur de QGIS histoire de se repérer.
```

### Mosaïquage

On va ici créer un fichier qui nous permettra de fusionner les dalles voulues pour notre carte.

> - Dans Qgis, on sélectionne les dalles de la région (rectangulaire) que l'on souhaite cartographier et on exporte la sélection au format csv qu'on nommera select.csv.

Reviewer: ```suggestion
- Dans QGIS, on sélectionne les dalles de la région (rectangulaire) que l'on souhaite cartographier et on exporte la sélection au format CSV qu'on nommera select.csv.
```

- On ouvre ce fichier dans LibreOfficeCalc (ou logiciel propriétaire équivalent) et on supprime l'entête des colonnes ainsi que toutes les colonnes sauf celle contenant le nom des tuiles.
- Dans la colonne adjacente on écrit cette formule :
    - Libre office :
    - =CONCAT(A1;".asc")
    - Excel :
    - =CONCATENER(A1;".asc")

> - On tire la formule et on remplace par les valeurs "en dur " avec un collage spécial

Reviewer: Qu'est ce que tu veux dire par "on tire la formule"?

Author: Appliquer la formule sur l'ensemble de la colonne concernée, je rends ça plus clair !

- Puis on supprime la colonne d'origine.

> - Enfin on change à la brutasse l'extension du fichier en .txt ce qui nous donne une fois ouvert :

Reviewer: ```suggestion
- Enfin on change à la brutasse l'extension du fichier en TXT ce qui nous donne une fois ouvert :
```
NB: 'harmonise le format typographique des types d'extensions pour être cohérent sur l'ensemble de l'article: j'ai choisi de les indiquer en majuscule et sans point devant (ASC, TXT) mais ce n'est pas forcément ce format là qu'il faut retenir :)

![exemple fichier tuile](https://cdn.geotribu.fr/img/articles-blog-rdp/articles/2024/gdal_qgis_blender/img2_tuiles.png){: .img-center loading=lazy }

> - On déplace ce fichier directement dans le répertoire contenant les tuilles asc téléchargées.

Reviewer: ```suggestion
- On déplace ce fichier directement dans le répertoire contenant les tuiles ASC téléchargées.
```

Et maintenant GDAL coeuravélesdoigts.

> Pour les non initiés à la ligne de commande, on se déplace dans un répertoire avec la commande cd (pour change directory). On peut soit passer un repertoire situé dans le repertoire courant, soit une adresse complète.

Reviewer: ```suggestion
Pour les non-initiés à la ligne de commande, on se déplace dans un répertoire avec la commande cd (pour _change directory_). On peut soit passer un répertoire situé dans le repertoire courant, soit une adresse complète.
```

Ex :

- sur Linux : `cd /home/nabuchodonosor/Documents`
- sur Windows : `cd C:\\\Users\nabuchodonosor\Documents`

> Sur windows pour changer de lecteur juste indiquer la lettre et les deux points, sans cd (exemple `D:`).

Reviewer: ```suggestion
Sur Windows, pour changer de lecteur, juste indiquer la lettre et les deux points, sans cd (exemple `D:`).
```

(on peut copier l'adresse dans l'explorateur et la coller pour se simplifier la vie)

Pour remonter d'un cran dans l'arboresence :
`cd ..`

> Vous vous souvenez de ces gens relous qui vous demandent des noms de dossiers/fichiers juste en alphanumériques et sans espaces mais avec des _ ? C'est pour ça.

Reviewer: Il n'y a pas un balisage pour faire un encadré façon "tips" ou "le saviez-vous"? Ce serait drôle d'encadrer cette phrase de cette manière :)

> Vous vous souvenez de ces gens relous qui vous demandent des noms de dossiers/fichiers juste en alphanumériques et sans espaces mais avec des _ ? C'est pour ça.

Reviewer: ```suggestion
Vous vous souvenez de ces gens relous qui vous demandent des noms de dossiers/fichiers juste en alphanumériques et sans espaces mais avec des _ ? C'est pour ça.
```

Il n'y a pas un balisage pour faire un encadré façon "tips" ou "le saviez-vous"? Ce serait drôle d'encadrer cette phrase de cette manière :)

Author: !!! tip  semble t-il (pouvez pas synchro vos languages histoire de toujours avoir les mêmes balises pour faire ça :P )

Les commandes GDAL sont accompagnées de `-` ou `--` et de lettres, ceci correspond aux options spécifiques du programme.

> On se déplace dans le dossier contenant les images ex :

Reviewer: ```suggestion
On se déplace dans le dossier contenant les images. Exemple :
```

![exemple fichier tuile](https://cdn.geotribu.fr/img/articles-blog-rdp/articles/2024/gdal_qgis_blender/img3_cd.png){: .img-center loading=lazy }

> Et maitenant on va utiliser [GDAL_merge.py](https://GDAL.org/programs/GDAL_merge.html), le programme de GDAL permettant de fusionner des images.

Reviewer: ```suggestion
Et maitenant on va utiliser [GDAL_merge.py](https://GDAL.org/programs/GDAL_merge.html), le programme de GDAL permettant de fusionner des images.
```

Peut-être transformer cette phrase qui commence par "Et maintenant" (comme celle de la ligne 63): Autant la changer en titre, non? Ça fera respirer et ça structure le texte

Author: Yep, j'ai reformulé

```bash
GDAL_merge.py -o mosaic.tif -co BIGTIFF=YES --optfile select.txt
```

> L'option`-o` permet de spécifier le nom du fichier de sortie (c'est donc obligatoire). On ne caressera jamais assez dans le sens du poil les gens derrière GDAL donc on dit que c'est très fort et ça reconnait le type de fichier désiré juste avec l'extension.

Reviewer: ```suggestion
- L'option`-o` permet de spécifier le nom du fichier de sortie (c'est donc obligatoire). On ne caressera jamais assez dans le sens du poil les gens derrière GDAL donc on dit que c'est très fort et ça reconnait le type de fichier désiré juste avec l'extension.
```
Pareil ici, structuration du contenu pour ajouter de la respiration :)

> `-co` correspond aux options spécifiques non pas du programme mais du driver du type de fichier (ici geotiff). BIGTIFF permet de faire des tif de plus de 4gb. Ce n'est pas mon cas ici mais ça ne coute rien de passer la commande par sécurité.

Reviewer: ```suggestion
`- -co` correspond aux options spécifiques non pas du programme mais du driver du type de fichier (ici geotiff). BIGTIFF permet de faire des tif de plus de 4gb. Ce n'est pas mon cas ici mais ça ne coute rien de passer la commande par sécurité.
```

> Il faut ensuite normalement spécifier un par un les noms de fichiers à fusionner ce qui serait fastidieux, mais l'option `--optfile` nous permet de passer un fichier contenant une liste, d'où les étapes précedantes !

Reviewer: ```suggestion
Il faut ensuite normalement spécifier un par un les noms de fichiers à fusionner, ce qui serait fastidieux, mais l'option `--optfile` nous permet de passer un fichier contenant une liste, d'où les étapes précedentes !
```

> Dans le repertoire vous trouverez votre geotiff mosaic.tif que je vous encourage à déplacer ailleurs pour s'y retrouver.

Reviewer: ```suggestion
Dans le répertoire, vous trouverez votre geotiff mosaic.tif que je vous encourage à déplacer ailleurs pour vous y retrouver.
```

> Dans ~~l'interface graphique de GDAL~~ Qgis cette étape est faisable via le menu raster -> Divers -> fusion (il faut au préalable charger l'ensemble des tuiles voulues dans qgis).

Reviewer: ```suggestion
Dans ~~l'interface graphique de GDAL~~ QGIS cette étape est faisable via le menu raster -> Divers -> fusion (il faut au préalable charger l'ensemble des tuiles voulues dans QGIS).
```

### Reprojection

> On va maintenant reprojeter (vous vous souvenez des fichiers asc ?) notre image avec [GDALwarp](https://GDAL.org/programs/GDALwarp.html) le programme servant à ... reprojeter.

Reviewer: ```suggestion
On va maintenant reprojeter (vous vous souvenez des fichiers ASC ?) notre image avec [GDALwarp](https://GDAL.org/programs/GDALwarp.html) le programme servant à... reprojeter.
```

Se déplacer dans le répertoire où vous avez placé l'image et :

```bash
GDALwarp -t_srs EPSG:2154 -r near -co BIGTIFF=YES mosaic.tif mosaicl93.tif
```

> - L'option `-t_srs` sert à indiquer le srid de sortie.

Reviewer: ```suggestion
- L'option `-t_srs` sert à indiquer le SRID de sortie.
```

> - `-r` permet de spécifier la méthode de rééchantillonage. Le choix dépasse le cadre de cet article mais sachez que des methodes avancées comme cubicspline ou lanczos peuvent donner des résultats "floutés" car modifiant la valeur des pixels selon des courbes. On va donc rester sur la méthode par défaut : nearest neighbour (plus proche voisin) pour éviter ceci.

Reviewer: ```suggestion
- `-r` permet de spécifier la méthode de rééchantillonnage. Le choix dépasse le cadre de cet article mais sachez que des methodes avancées comme `cubicspline` ou `lanczos` peuvent donner des résultats "floutés" car modifiant la valeur des pixels selon des courbes. On va donc rester sur la méthode par défaut : `nearest neighbour` (plus proche voisin) pour éviter ceci.
```

> Pour celles et ceux à l'aise avec GDAL, vous pouvez compresser les images pour réduire la taille des fichiers de sortie au moins en deflate (pour les autres, ça se fait en passant une seconde option pour le driver de type de fichier : `-co COMPRESS=methode`, la plus courrament utilisée étant `-co COMPRESS=DEFLATE` pour s'assurer de la compatibilité avec tous les systèmes / logiciels).

Reviewer: ```suggestion
Pour celles et ceux à l'aise avec GDAL, vous pouvez compresser les images pour réduire la taille des fichiers de sortie au moins en `deflate` (pour les autres, ça se fait en passant une seconde option pour le driver de type de fichier : `-co COMPRESS=methode`, la plus couramment utilisée étant `-co COMPRESS=DEFLATE` pour s'assurer de la compatibilité avec tous les systèmes / logiciels).
```

> Sinon, ça se fait avec raster -> Projection -> Assigner une projection dans Qgis.

Reviewer: ```suggestion
Sinon, dans QGIS, ça se fait avec Raster -> Projection -> Assigner une projection.
```

### Fausser les données

> Oui. Nous allons commettre ceci. Ne sortez pas les bidons d'essence tout de suite s'il vous plait. Pour ce que nous allons en faire, Blender n'accepte que les images en entier 16 bits non signés (UInt16, on y reviendra), donc une plage de valeur pour les pixels comprise entre 0 et 65 535. Mais sans virgules, ce que contient notre raster initial donc on perdrait du détail. L'idée est donc de réattribuer aux pixels de notre image des valeurs sur l'ensemble de cette plage, ceci pour bénéficier de l'entièreté de cette finesse.

Reviewer: ```suggestion
Oui. Nous allons commettre ceci. Ne sortez pas les bidons d'essence tout de suite s'il vous plaît. Pour ce que nous allons en faire, Blender n'accepte que les images en entier 16 bits non signés (UInt16, on y reviendra), donc une plage de valeur pour les pixels comprise entre 0 et 65 535. Mais sans virgules, ce que contient notre raster initial donc on perdrait du détail. L'idée est donc de réattribuer aux pixels de notre image des valeurs sur l'ensemble de cette plage, ceci pour bénéficier de l'entièreté de cette finesse.
```

> On fait ça avec [GDAL_calc.py](https://GDAL.org/programs/GDAL_calc.html) mais on peut aussi rester simple et faire ça avec la calculatrice raster de Qgis.

Reviewer: ```suggestion
On fait ça avec [GDAL_calc.py](https://GDAL.org/programs/GDAL_calc.html) mais on peut aussi rester simple et faire ça avec la calculatrice raster de QGIS.
```

> Tout d'abord, on normalise notre raster car certaines valeurs peuvent êtres en dessous de 0 (notamment quand il y a de l'eau). On va donc les mettres à 0.

Reviewer: ```suggestion
Tout d'abord, on normalise notre raster car certaines valeurs peuvent êtres en dessous de 0 (notamment quand il y a de l'eau). On va donc les mettre à 0.
```

```bash
GDAL_calc.py -A moasicl93.tif --outfile=mosaic_cut.tif --calc="A*(A>=0)"
```

> Avec GDAL_calc.py les maths doivent être "[numpy](https://numpy.org/) style".

Reviewer: ```suggestion
Avec `GDAL_calc.py`, les maths doivent être "[numpy](https://numpy.org/) style".
```

- `-A` permet d'attribuer un raster à une lettre qui sera ensuite utilisée dans la formule.
- `--outfile` le nom du fichier de sortie (ça dépend des programmes GDAL, souvent c'est la position du nom qui fait foi dans la commande).

> - `--calc` la formule de calcul, entourée de "".

Reviewer: ```suggestion
- `--calc` la formule de calcul, entourée de `""`.
```

> Ou dans qgis : raster -> calculatrice raster puis

Reviewer: ```suggestion
Ou dans QGIS : raster -> calculatrice raster puis
```

![calc_raster](https://cdn.geotribu.fr/img/articles-blog-rdp/articles/2024/gdal_qgis_blender/img4_calc.png)

if (condition, valeur si vrai, valeur si faux)

On regarde les valeurs minimum et maximum de notre raster. Soit dans les propriétés de la couche qgis, soit avec [GDALinfo](https://GDAL.org/programs/GDALinfo.html).

```bash
GDALinfo -mm mosaic_cut.tif
```

- l'option `-mm` permet de forcer le calcul des valeurs min/max qui n'est pas donné par défaut par GDALinfo. Elles apparaitrons tout en bas (dans mon cas, 0 et 196,8).

![La](https://cdn.geotribu.fr/img/articles-blog-rdp/articles/2024/gdal_qgis_blender/img5_minmax.png){: .img-center loading=lazy }

Le calcul à faire pour réaffecter nos pixels est le suivant :
(Valeur pixel – min) ÷ (max – min) * 65535

Dans mon cas celà donne :

```bash
GDAL_calc.py -A mosaic_cut.tif --outfile=mosaic_rescale.tif --calc="(A - 0) / (196.8 - 0) * 65535"
```

Dans la calculatrice raster de Qgis celà donnerai :
("mosaic_cut@1" - 0) / (196.8 - 0) * 65535.

> Enfin, on va convertir notre raster en UInt16 pour l'import dans Blender. C'est cette opération qui n'est pas réalisable dans Qgis sans circonvolutions. Donc on sort [GDAL_translate](https://GDAL.org/programs/GDAL_translate.html), le programme qui sert à faire des conversions.

Reviewer: ```suggestion
Enfin, on va convertir notre raster en UInt16 pour l'importer dans Blender. C'est cette opération qui n'est pas réalisable dans QGIS sans circonvolutions. Donc on sort [gdal_translate](https://GDAL.org/programs/GDAL_translate.html), le programme qui sert à faire des conversions.
```

```bash
GDAL_translate -ot UInt16 mosaic_rescale.tif mnt.tif
```

- `-ot` est l'option qui permet de forcer le type de données de sortie.
- les fichiers d'entrée et de sortie se précisent par leur position dans la commande comme précisé dans la [**documentation**](https://GDAL.org/index.html) très complète de GDAL que je m'efforce de vous inciter à consulter depuis le début en vous en spammant le lien le plus de fois possible.

## Petite présentation de Blender et configuration

Pour télécharger Blender ça se passe [ici](https://www.blender.org/download/). C'est un logiciel libre donc pas d'inquiétudes. (sur Linux il est aussi présent sur [Flatpak](https://flathub.org/apps/org.blender.Blender)).

> Faire le tour de Blender serait bien sur beaucoup trop ambiteux ici, mais voici quelques indications de base pour celles et ceux qui n'ont jamais ouvert le logiciel.

Reviewer: ```suggestion
Faire le tour de Blender serait bien sûr beaucoup trop ambiteux ici, mais voici quelques indications de base pour celles et ceux qui n'ont jamais ouvert le logiciel.
```

> Blender fonctionne sur un principe d'environnement en fonction de la tache que vous êtes en train de réaliser (modéliser, travailler sur les textures ...). Pour passer de l'un à l'autre, on clique ici :

Reviewer: ```suggestion
Blender fonctionne sur un principe d'environnement en fonction de la tâche que vous êtes en train de réaliser (modéliser, travailler sur les textures ...). Pour passer de l'un à l'autre, on clique ici :
```

> ![La](https://cdn.geotribu.fr/img/articles-blog-rdp/articles/2024/gdal_qgis_blender/img6_viewports.png){: .img-center loading=lazy }

Reviewer: ```suggestion
![Là](https://cdn.geotribu.fr/img/articles-blog-rdp/articles/2024/gdal_qgis_blender/img6_viewports.png){: .img-center loading=lazy }
```

> 3D Viewport est la vue 3d par défaut.

Reviewer: ```suggestion
3D Viewport est la vue 3D par défaut.
```

> Les boutons situés ici permettent de changer le mode d'affichage des objets 3d (fil de fer, materiaux, rendu ...).

Reviewer: ```suggestion
Les boutons situés ici permettent de changer le mode d'affichage des objets 3D (fil de fer, materiaux, rendu...).
```

> ![La](https://cdn.geotribu.fr/img/articles-blog-rdp/articles/2024/gdal_qgis_blender/img7_render.png){: .img-center loading=lazy }

Reviewer: ```suggestion
![Là](https://cdn.geotribu.fr/img/articles-blog-rdp/articles/2024/gdal_qgis_blender/img7_render.png){: .img-center loading=lazy }
```

> (vous pouvez essayer avec le cube présent par défaut).

Reviewer: ```suggestion
(Vous pouvez essayer avec le cube présent par défaut).
```

### Réglages préalables (moteur de rendu et carte graphique)

En cliquant là :

> ![La](https://cdn.geotribu.fr/img/articles-blog-rdp/articles/2024/gdal_qgis_blender/img8_engine.png){: .img-center loading=lazy }

Reviewer: ```suggestion
![Là](https://cdn.geotribu.fr/img/articles-blog-rdp/articles/2024/gdal_qgis_blender/img8_engine.png){: .img-center loading=lazy }
```

Vous pouvez changer le moteur de rendu utilisé entre Eevee et Cycles. Eevee est plus couramment utilisé pour du dynamique, et Cycles pour du rendu statique (notre cas). Attention, Cycles est plus gourmant en ressources. **Important :** Choisissez aussi le "Feature Set" Expérimental (nous en aurons besoin). Enfin, si vous faites des choix de vie douteux comme moi et que votre carte graphique est puissante, passez "Device" en GPU compute.

> En fonction de votre carte graphique, vous pouvez aussi faire un tour par le menu edit -> preferences -> system et choisir en fonction de votre crémerie ce qui sera utilisé par Cycles. Choisir [OptiX](https://fr.wikipedia.org/wiki/OptiX) chez Nvidia / [HIP](https://rocm.docs.amd.com/projects/HIP/en/latest/index.html) chez AMD si votre configuration matérielle le supporte (hey, vous venez sur un tuto 3d, il faut s'attendre à ce genre de phrases !).

Reviewer: ```suggestion
En fonction de votre carte graphique, vous pouvez aussi faire un tour par le menu edit -> preferences -> system et choisir en fonction de votre crémerie ce qui sera utilisé par Cycles. Choisir [OptiX](https://fr.wikipedia.org/wiki/OptiX) chez Nvidia / [HIP](https://rocm.docs.amd.com/projects/HIP/en/latest/index.html) chez AMD si votre configuration matérielle le supporte (hey, vous venez sur un tuto 3D, il faut s'attendre à ce genre de phrases !).
```

Toujours là :

> ![La](https://cdn.geotribu.fr/img/articles-blog-rdp/articles/2024/gdal_qgis_blender/img9_sampling.png){: .img-center loading=lazy }

Reviewer: ```suggestion
![Là](https://cdn.geotribu.fr/img/articles-blog-rdp/articles/2024/gdal_qgis_blender/img9_sampling.png){: .img-center loading=lazy }
```

> En descendant vous verez Sampling. Ces options permettent de configurer le nombre de passages qu'effectuera le moteur de rendu en changeant la valeur max samples. Je vous conseille de configurer Viewport (quand vous serez en train de travailler mais en affichant quelque chose proche du rendu) avec une valeur basse pour gagner du temps, et Render avec une valeur haute pour avoir un beau rendu final. Activer l'option Denoise dans les deux cas qui permet d'enlever du "bruit" sur les rendus. Sur l'image vous verez ma proposition de paramètrage.

Reviewer: ```suggestion
En descendant vous verrez "Sampling". Ces options permettent de configurer le nombre de passages qu'effectuera le moteur de rendu en changeant la valeur `max samples`. Je vous conseille de configurer Viewport (quand vous serez en train de travailler mais en affichant quelque chose proche du rendu) avec une valeur basse pour gagner du temps, et Render avec une valeur haute pour avoir un beau rendu final. Activer l'option Denoise dans les deux cas qui permet d'enlever du "bruit" sur les rendus. Sur l'image vous verrez ma proposition de paramétrage.
```

> Enfin la chose la plus importante, Blender fonctionne beaucoup avec les raccourcis clavier. Pour ajouter un objet : `shift + A`.

Reviewer: ```suggestion
Enfin la chose la plus importante, Blender fonctionne beaucoup avec les raccourcis clavier. Pour ajouter un objet : `Shift + A`.
```

---

> title: "Réaliser des cartes avec Blender"
> authors:

Reviewer 2: Pour info, tu peux ajouter un court sous-titre qui apparaît dans la barre de navigation de gauche, dans l'image d'en-tête si auto-générée, la liste des articles...

```suggestion
title: "Réaliser des cartes avec Blender"
subtitle: 
authors:
```

    - Thomas Szczurek-Gayant
categories:
    - article
comments: true
date: 2024-04-07
description: "Technique permettant de réaliser des cartes de relief avec le logiciel libre 3d Blender, ainsi qu'un petit tutoriel gdal"
icon: simple/blender

> image: https://github.com/thomas-szczurek/images/blob/main/title.png

Reviewer 2: Toutes les images doivent être uploadées sur notre CDN (sauf si elles sont trop lourdes) et celle pour l'en-tête doit respecter un certain ratio de dimensions.

Voir :

- https://contribuer.geotribu.fr/guides/metadata_yaml_frontmatter/#syntaxe pour les consignes sur l'image d'en-tête
- https://contribuer.geotribu.fr/guides/cdn-images-hebergement/

Je t'envoie les accès au CDN par mail.

Reviewer 2: Tu peux aussi ne rien mettre dans ce champ, une image sera alors automatiquement généréé. C'est ce que je te suggère ici (mais c'est à ta convenance) :

```suggestion
image:
```

Sur ces considérations on chargera plutôt la couche "dalles" [dans le format dont il ne faut pas prononcer le nom](http://switchfromshapefile.org/) située dans le dossier "3_SUPPLEMENTS_LIVRAISON" de l'archive et on active un fond de plan open street map sous "xyz tiles" de l'explorateur de Qgis histoire de se repérer.

### Mosaïquage

On va ici créer un fichier qui nous permettra de fusionner les dalles voulues pour notre carte.

- Dans Qgis, on sélectionne les dalles de la région (rectangulaire) que l'on souhaite cartographier et on exporte la sélection au format csv qu'on nommera select.csv.
- On ouvre ce fichier dans LibreOfficeCalc (ou logiciel propriétaire équivalent) et on supprime l'entête des colonnes ainsi que toutes les colonnes sauf celle contenant le nom des tuiles.
- Dans la colonne adjacente on écrit cette formule :

```
# Libre office
=CONCAT(A1;".asc")
# Excel
=CONCATENER(A1;".asc")
```

- On tire la formule et on remplace par les valeurs "en dur " avec un collage spécial
- Puis on supprime la colonne d'origine.
- Enfin on change à la brutasse l'extension du fichier en .txt ce qui nous donne une fois ouvert :

> ![exemple fichier tuile](https://github.com/thomas-szczurek/images/blob/main/img2_tuiles.png)

Reviewer 2: Ajouter les balises de centrage et chargement optimisé (cf https://contribuer.geotribu.fr/guides/image/)

```suggestion
![exemple fichier tuile](https://github.com/thomas-szczurek/images/blob/main/img2_tuiles.png){: .img-center loading=lazy }
```

- On déplace ce fichier directement dans le répertoire contenant les tuilles asc téléchargées.

Et maintenant GDAL coeuravélesdoigts.

Pour les non initiés à la ligne de commande, on se déplace dans un répertoire avec la commande cd (pour change directory). On peut soit passer un repertoire situé dans le repertoire courant, soit une adresse complète.

Ex :

> cd /home/nabuchodonosor/Documents (Linux)
> cd C:\\\Users\nabuchodonosor\Documents (Windows)

Reviewer 2: Ajouter les balises de code :

```suggestion

- sur Linux : `cd /home/nabuchodonosor/Documents`
- sur Windows : `cd C:\\\Users\nabuchodonosor\Documents`
```

---
title: "Réaliser des cartes avec Blender"

> subtitle:

Author: je laisse les calembours aux spécialistew !

Enfin la chose la plus importante, Blender fonctionne beaucoup avec les raccourcis clavier. Pour ajouter un objet : `Shift + A`.

Pour se déplacer, appuyez sur le bouton central de votre souris (le clic de la molette) pour tourner, maintenez shift en plus pour translater ou ctrl pour zoomer.

## Modéliser le relief

Tout d'abord, on retire cube par défaut en cliquant dessus pour le sélectionner puis en appuyant sur `suppr` de votre clavier et on ajoute un plan à notre scène. `shift + A` -> mesh -> plane. Ne pas supprimer la caméra et la source de lumière (les autres trucs présents dans la scène par défaut).

Par défaut les objets apparaissent sous le "curseur 3D" mais on va déplacer notre plan pour le mettre aux coordonnées x 0 y 0 z 0 pour se simplifier la vie si ce n'est pas le cas.

On clique sur notre plan pour le sélectionner (il se détoure alors en orange) et on se rend là :

![La](https://cdn.geotribu.fr/img/articles-blog-rdp/articles/2024/gdal_qgis_blender/img10_loc.png){: .img-center loading=lazy }

On modifie ensuite les valeurs de `Location` x y et z à 0 si ce n'est pas le cas.

> Je vous ai déjà parlé de mon amour de GDAL ? Et bien on ve le ressortir avec encore une fois la commande `GDALinfo`. En effet maintenant on va donner à notre plan les dimensions de notre raster d'elevation.

Reviewer: ```suggestion
Je vous ai déjà parlé de mon amour de GDAL ? Et bien on ve le ressortir avec encore une fois la commande `gdalinfo`. En effet maintenant on va donner à notre plan les dimensions de notre raster d'elevation.
```

![taille raster](https://cdn.geotribu.fr/img/articles-blog-rdp/articles/2024/gdal_qgis_blender/img11_size.png){: .img-center loading=lazy }

La ligne Size vous donne la taille en pixels de votre mnt.

> Toujours dans ce panneau de Blender, on modifie `scale` pour faire correspondre. Indiquer x 8000 / y 7000 (mon cas) donnerai un plan beaucoup trop grand, l'important est de garder le ratio. J'indique donc x 0.8 et y 0.7 comme valeur pour scale.

Reviewer: ```suggestion
Toujours dans ce panneau de Blender, on modifie `scale` pour faire correspondre. Indiquer x 8000 / y 7000 (mon cas) donnerait un plan beaucoup trop grand, l'important est de garder le ratio. J'indique donc x 0.8 et y 0.7 comme valeur pour scale.
```

> Toujours dans ce panneau de Blender, on modifie `scale` pour faire correspondre. Indiquer x 8000 / y 7000 (mon cas) donnerai un plan beaucoup trop grand, l'important est de garder le ratio. J'indique donc x 0.8 et y 0.7 comme valeur pour scale.

Reviewer: ```suggestion
Toujours dans ce panneau de Blender, on modifie `scale` pour faire correspondre. Indiquer x 8000 / y 7000 (mon cas) donnerai un plan beaucoup trop grand, l'important est de garder le ratio. J'indique donc x 0.8 et y 0.7 comme valeur pour `scale`.
```

> Je ne répeterai pas cette information mais pensez à sauvegarder ! File -> save as et enregistrez votre espace de travail au format `.blend.`

Reviewer: ```suggestion
Je ne répeterai pas cette information mais pensez à sauvegarder ! `File > Save as`, puis enregistrez votre espace de travail au format `.blend.`
```

### Déformer le plan

C'est ici qu'on va commencer a faire des trucs un peu compliqués.

> Dans le monde réel, les substances telles que le bois, la roche, le verre ... semblent différentes les unes des autres car elles ont des couleurs, des textures, une rugosité différentes. Blender est conçu pour simuler ces variations en permettant d'attribuer des propriétés reflétant ceci aux `materiaux` attribués aux objets. Les moteurs de rendus calculent ensuite l'apparence des objets ainsi que la dispersion et les rebonds de la lumière en fonction de ces `matériaux`.

Reviewer: ```suggestion
Dans le monde réel, les substances telles que le bois, la roche, le verre ... semblent différentes les unes des autres car elles ont des couleurs, des textures, une rugosité différentes. Blender est conçu pour simuler ces variations en permettant d'attribuer des propriétés reflétant ceci aux `matériaux` attribués aux objets. Les moteurs de rendus calculent ensuite l'apparence des objets ainsi que la dispersion et les rebonds de la lumière en fonction de ces `matériaux`.
```

C'est `matériaux`ou `materiaux` dans Blender? (Oui, je coupe les cheveux en quatre)

> Pour l'instant, notre plan ne possèque aucun matériau, ce pourquoi il apparait gris mat. Si vous effectuez un rendu en cliquant sur render -> render image, ou en choissant le mode de visualisation "rendu" dans le 3d view port, vous vous en rendrez compte.

Reviewer: ```suggestion
Pour l'instant, notre plan ne possèque aucun matériau, ce pourquoi il apparaît gris mat. Si vous effectuez un rendu en cliquant sur `Render -> Render image`, ou en choissant le mode de visualisation "Rendu" dans le 3D view port, vous vous en rendrez compte.
```

> Pour affecter un matériau au plan, selectionnez le puis cliquez sur cette icone :

Reviewer: ```suggestion
Pour affecter un matériau au plan, selectionnez-le puis cliquez sur cette icone :
```

![setup materials](https://cdn.geotribu.fr/img/articles-blog-rdp/articles/2024/gdal_qgis_blender/img12_material.png){: .img-center loading=lazy }

> Puis cliquez sur `New`. Blendez créera alors un nouveau materiau nommé Material.001. Vous pouvez le renommer si vous le désirez.

Reviewer: ```suggestion
Puis cliquez sur `New`. Blendez créera alors un nouveau matériau nommé `Material.001`. Vous pouvez le renommer si vous le désirez.
```

Pour l'instant nous ne modifierons rien mais voici quelques explications :

- L'option `Surface` indique "Principled BSDF". Ce n'est pas référence à une pratique étrange du yoga mais à un `shader` utilisant une ["Bidirectional Scattering Distribution Function"](https://en.wikipedia.org/wiki/Bidirectional_scattering_distribution_function). Les `shaders` indiquent à Blender la manière dont la lumière rebondi sur les objets. Chaque `shader` présent dans cette liste représente un modèle mathématique d'interaction de la lumière avec l'objet. Utiliser autre chose que Principled BSDF permet de faire des objets qui ressemblerons à du verre, du coton... mais ce n'est pas ce que nous voulons ici donc on va laisser ce paramètre tranquille.

> - Il y a beaucoup d'options pour ce `shader` : la rugosité, l'incide de refraction, la couleur ... mais pour le moment nous voulons un plan plat et réaliste, donc on touche à rien. On y reviendra plus tard.

Reviewer: ```suggestion
- Il y a beaucoup d'options pour ce `shader` : la rugosité, l'incide de refraction, la couleur ... mais pour le moment nous voulons un plan plat et réaliste, donc on ne touche à rien. On y reviendra plus tard.
```

> Les `matériaux` ne servent pas qu'à assigner une couleur ou une interaction avec la lumière. Ils peuvent aussi posséder un `displacement` / `déplacement`. Soit une déformation de leur surface. Et c'est comme ça que nous allons transformer notre plan en un modèle de relief réaliste.

Reviewer: ```suggestion
Les `matériaux` ne servent pas qu'à assigner une couleur ou une interaction avec la lumière. Ils peuvent aussi posséder un `displacement` / `déplacement`, soit une déformation de leur surface. C'est comme ça que nous allons transformer notre plan en un modèle de relief réaliste.
```

> Changez l'environnement de travail pour passer de "3D View port" à "Shader editor" (voir la petite présentation de Blender plus haut pour voir comment faire).

Reviewer: ```suggestion
Changez l'environnement de travail pour passer de "3D Viewport" à "Shader Editor" (voir la petite présentation de Blender plus haut pour voir comment faire).
```

> Cette interface permet de régler les paramètres du `matériau` comme précédemment, mais bien plus encore. Faites bien attention à bien sélectionner votre plan en cliquant là :

Reviewer: ```suggestion
Cette interface permet de régler les paramètres du `matériau` comme précédemment, mais bien plus encore. Faîtes bien attention à bien sélectionner votre plan en cliquant là :
```

![selectionner le plan](https://cdn.geotribu.fr/img/articles-blog-rdp/articles/2024/gdal_qgis_blender/img13_explorer.png){: .img-center loading=lazy }

> La manière dont est rendu votre matériau est présentée sous forme d'un diagramme. Vous pouvez cliquer sur chacune des boites pour les déplacer, ainsi que vous déplacer dans la vue comme dans le 3D View Port.

Reviewer: ```suggestion
La manière dont est rendu votre matériau est présentée sous forme d'un diagramme. Vous pouvez cliquer sur chacune des boîtes pour les déplacer, ainsi que vous déplacer dans la vue comme dans le 3D View Port.
```

> Chaque boite est un `node` et vous verrez une ligne reliant Principle BSDF à la "Surface" de notre `matériau` final, indiquant qu'il est utilisé pour déterminer son apparence. Vous verrez que vous pouvez aussi relier quelque chose qui donnera a la surface un `displacement`.

Reviewer: ```suggestion
Chaque boîte est un `node` et vous verrez une ligne reliant "Principle BSDF" à la "Surface" de notre `matériau` final, indiquant qu'il est utilisé pour déterminer son apparence. Vous verrez que vous pouvez aussi relier quelque chose qui donnera à la surface un `displacement`.
```

> Depuis la barre de menu située au dessus de l'écran, choisissez Add -> Texture -> Image texture.

Reviewer: ```suggestion
Depuis la barre de menu située au dessus de l'écran, choisissez `Add > Texture > Image Texture`.
```

![ajouter texture](https://cdn.geotribu.fr/img/articles-blog-rdp/articles/2024/gdal_qgis_blender/img14_addtexture.png){: .img-center loading=lazy }

> Une nouvelle boite apparait, positionnez la où bon vous semble. Dans le language de Blender une `texture` est une image ou un motif qui sera appliqué aux `matériaux` pour changer leur apparence. On pourrait ainsi charger une image de grain de bois pour faire ressembler notre plan à du bois. Mais ces texture peuvent aussi être utilisées pour générer un `displacement`.

Reviewer: ```suggestion
Une nouvelle boite apparait. Positionnez-la où bon vous semble. Dans le language de Blender, une `texture` est une image ou une motif qui sera appliquée aux `matériaux` pour changer leur apparence. On pourrait ainsi charger une image de grain de bois pour faire ressembler notre plan à du bois. Mais ces texture peuvent aussi être utilisées pour générer un `displacement`.
```

> - Cliquez sur `Open` dans cette boite et indiquez votre mnt.tif.

Reviewer: ```suggestion
- Cliquez sur `Open` dans cette boite et indiquez votre `mnt.TIF`.
```

- Cliquez sur le petit cercle à côté de `Color` en maintenant le bouton gauche de la souris appuyé.
- Et relier à `Displacement`
- Tant que nous y sommes, changez l'option `Linear` pour `Smart`. Ceci change l'interpolation de la texture pour être un peu plus jolie.

![ajouter texture](https://cdn.geotribu.fr/img/articles-blog-rdp/articles/2024/gdal_qgis_blender/img15_displacement.png){: .img-center loading=lazy }

> Vous pouvez faire un essai de rendu en cliquant sur Render -> render image dans le menu tout en haut.

Reviewer: ```suggestion
Vous pouvez faire un essai de rendu en cliquant sur `Render -> Render Image` dans le menu tout en haut.
```

![ajouter texture](https://cdn.geotribu.fr/img/articles-blog-rdp/articles/2024/gdal_qgis_blender/img16_1strender.png){: .img-center loading=lazy }

> Pour repprendre les mots de l'auteur originel "Tout ceci est bâti sur un mensonge". Pour le moment Blender ne fait que **simuler** la déformation en utilisant une technique nommée _bump mapping_. C'est plus rapide à calculer mais pas aussi réaliste. Cette technique ne fait que donner une _apparence_ de profondeur, mais elle n'est pas réelle : aucune ombre n'est projetée et la lumière n'interagit pas avec.

Reviewer: ```suggestion
Pour reprendre les mots de l'auteur originel : "Tout ceci est bâti sur un mensonge". Pour le moment Blender ne fait que **simuler** la déformation en utilisant une technique nommée _bump mapping_. C'est plus rapide à calculer mais pas aussi réaliste. Cette technique ne fait que donner une _apparence_ de profondeur, mais elle n'est pas réelle : aucune ombre n'est projetée et la lumière n'interagit pas avec.
```

On va faire beaucoup mieux.

### Subdiviser la surface

Pour l'instant notre plan est trop simpliste, car uniquement composé de 4 `vertices`/`vertex`/`noeud` whatever vous voyez de quoi je parle.

Ces `vertices` sont ce que Blender utilise pour déformer le terrain et pour le moment Blender ne pourrait utiliser que les quatre coins de l'image.

> Pour l'instant ces derniers ne sont pas utilsés, Blender n'a fait que _peindre_ le relief sur le plan avec des nuances de gris. On pourrait ajouter des `vertices` à la main mais on va utiliser une autre astuce. Les `modificateurs`.

Reviewer: ```suggestion
Pour l'instant ces derniers ne sont pas utilisés, Blender n'a fait que _peindre_ le relief sur le plan avec des nuances de gris. On pourrait ajouter des `vertices` à la main, mais on va utiliser une autre astuce. Les `modificateurs`.
```

> - Retournez dans le 3D View Port

Reviewer: ```suggestion
- Retournez dans le `3D Viewport`
```

> - Toujours bien penser à sélectionner le plan et cliquez sur l'icone en forme de clef à molette.

Reviewer: ### 
```suggestion
- Toujours bien penser à sélectionner le plan et cliquer sur l'icône en forme de clef à molette.
```

![modifiers](https://cdn.geotribu.fr/img/articles-blog-rdp/articles/2024/gdal_qgis_blender/img17_modifiers.png){: .img-center loading=lazy }

> - Et maintenant cliquez sur "Add modifier"

Reviewer: ```suggestion
- Et maintenant, cliquer sur "Add Modifier"
```

- L'apparence de ce menu a été modifiée dans la toute dernière version de Blender mais l'idée est de choisir le groupe "Generate" puis ["Subdivision Surface"](https://docs.blender.org/manual/fr/4.1/modeling/modifiers/generate/subdivision_surface.html).

> Apparemant la communauté Blender surnomme ce modificateur "subsurf" ou "subdiv". Ne me demandez pas. Il est très utilisé et permet de dire à Blender d'ajouter du détail à un objet (ainsi il permet de faire des objets très arrondis en travaillant avec des `mesh` simples).

Reviewer: ```suggestion
Apparemment, la communauté Blender surnomme ce modificateur `subsurf` ou `subdiv`. Ne me demandez pas. Il est très utilisé et permet de dire à Blender d'ajouter du détail à un objet (ainsi il permet de faire des objets très arrondies en travaillant avec des `mesh` simples).
```

Dans notre cas on va l'utilier pour faire croire à Blender que notre objet est composé de très nombreux morceaux afin de simuler beaucoup de `vertices`.

- Deux algorithmes sont possibles et celui que nous voulons ici est `simple`.

> - Si vous avez bien écouté au fond de la salle, vous verrez un bouton "adaptive subdivision" qu'il faut activer, sinon retournez sur la partie de présentation / configuration de Blender pour passez le feature set de Cycles à expérimental. Vous savez, celle ou j'avais indiqué "important". En gras.

Reviewer: ```suggestion
- Si vous avez bien écouté au fond de la salle, vous verrez un bouton "adaptive subdivision" qu'il faut activer, sinon retournez sur la partie de présentation / configuration de Blender pour passez le feature set de `Cycles` à `Expérimental`. Vous savez, celle ou j'avais indiqué "important". En gras.
```

- Cette option indique à Blender de ne pas subdiviser la surface par un nombre défini mais là où _le plus de détail est necessaire_ en fontion de la taille de votre image et de la position de la caméra.

> - Ce `modificateur` est non destructif, il ne modifie par réellement l'objet. C'est une pile qui s'ajoute _par dessus_ alors qu'il est toujours constitué de quatre `vertices`. Blender ne subdivisera le plan que temporairement lors du rendu.

Reviewer: ```suggestion
- Ce `modificateur` est non-destructif, il ne modifie par réellement l'objet. C'est une pile qui s'ajoute _par dessus_ alors qu'il est toujours constitué de quatre `vertices`. Blender ne subdivisera le plan que temporairement lors du rendu.
```

---
title: Réaliser des cartes avec Blender
subtitle: Blindé jusqu’au relief
authors:
    - Thomas Szczurek-Gayant
categories:
    - article
comments: true
date: 2024-04-07
description: Technique permettant de réaliser des cartes de relief avec le logiciel libre 3D Blender, ainsi qu'un petit tutoriel GDAL.
icon: simple/blender
image:
license: default
robots: index, follow
tags:
    - 3D
    - Blender
    - cartographie
    - GDAL
---

> # Réaliser des cartes avec Blender

Reviewer 3: Attention au poids des images dans l'articles. Elles sont souvent > à 1.5mo

```bash
gdal_translate -ot UInt16 mosaic_rescale.tif mnt.tif
```

- `-ot` est l'option qui permet de forcer le type de données de sortie.
- les fichiers d'entrée et de sortie se précisent par leur position dans la commande comme précisé dans la [**documentation**](https://GDAL.org/index.html) très complète de GDAL que je m'efforce de vous inciter à consulter depuis le début en vous en spammant le lien le plus de fois possible.

## Petite présentation de Blender et configuration

Pour télécharger Blender ça se passe [ici](https://www.blender.org/download/). C'est un logiciel libre donc pas d'inquiétudes. (sur Linux il est aussi présent sur [Flatpak](https://flathub.org/apps/org.blender.Blender)).

Faire le tour de Blender serait bien sûr beaucoup trop ambiteux ici, mais voici quelques indications de base pour celles et ceux qui n'ont jamais ouvert le logiciel.

Blender fonctionne sur un principe d'environnement en fonction de la tâche que vous êtes en train de réaliser (modéliser, travailler sur les textures ...). Pour passer de l'un à l'autre, on clique ici :

> ![Là](https://cdn.geotribu.fr/img/articles-blog-rdp/articles/2024/gdal_qgis_blender/img6_viewports.png){: .img-center loading=lazy }

Reviewer 3: Suggestion, plutôt que de mettre des flèches à la "Paint", peut utiliser libre office ou power point pour coller l'image et faire des flèches plus sympas...

Author: > Suggestion, plutôt que de mettre des flèches à la "Paint", peut utiliser libre office ou power point pour coller l'image et faire des flèches plus sympas...

Salut !
Je peux peut-être regarder pour faire ça mais à condition que j'arrive à retoucher les images déjà crées car je ne possède plus les versions d'origine sans flèches et j'ai moyennement envie de me refaire les plus de 40 captures ;) (mais ça pourrai être l'occasion de voir cette question de poids des images)

Pour définir l'angle d'incidence des rayons lumineux il faut se rendre dans les propriétés de l'objet (attention à bien le sélectionner) et changer sa rotation (pour les sources de lumière de type soleil).

Indiquer 0 pour x, 45 pour y et 135 pour z.

![réglages lumière](https://cdn.geotribu.fr/img/articles-blog-rdp/articles/2024/gdal_qgis_blender/img34_light3.png){: .img-center loading=lazy }

La valeur de y contrôle l'angle par rapport à l'horizon, et celle de z la direction de provenance. Une valeur de 135 vous donne une lumière qui arrive du Nord Ouest (en haut à gauche quoi), ce à quoi l'oeil humain est habitué pour une carte. Mettre 225 donnerai l'impression que notre relief est inversé.

Le tout dernier réglage à faire est de régler la taille de notre soleil. On reclique sur la petite ampoule verte et on regarde le paramètre Angle, mal nommé puisqu'il correspond au [diamètre angulaire](https://fr.wikipedia.org/wiki/Taille_apparente). Changer sa valeur régle la _douceur_ de la lumière. Avec une valeur basse par défault, notre lumière nous donne un relief lunaire, essayez plutôt une valeur de 90 pour quelque chose de plus doux.

> ![réglages lumière](https://cdn.geotribu.fr/img/articles-blog-rdp/articles/2024/gdal_qgis_blender//img35_light4.png){: .img-center loading=lazy }

Reviewer 3: Bizarre car cette image (et la suivante) sur le site de preview ne s'affiche pas.
Pourtant elles sont bien sur le CDN
